### PR TITLE
Fix disk writer should get a new hash for each file

### LINKED ISSUE #1159: Graveler's sstable manager share hash
Graveler's sstable manager holds hash which holds a state and share it each NewDiskWriter.
Because hash holds a state this can't be share.

### PR DESCRIPTION
**Pull request by nopcoder**
Fix #1159

### CODE REVIEW

**Review by arielshaqed — APPROVED** [review]
Good catch!

A unit test cannot easily detect issues like this (loss of catalog dedupe).  Wondering how to automate testing this in an ongoing manner.  Best I can see is to run specific import scenarios and track disk usage.

File: graveler/sstable/manager.go [hunk 1]
```diff
@@ -16,10 +16,10 @@ type Manager struct {
 	fs    pyramid.FS
 	// TODO(ariels): Replace with loggers constructed from context.
 	logger logging.Logger
-	hash   hash.Hash
+	hash   crypto.Hash
```
Comment (arielshaqed) [inline graveler/sstable/manager.go:19]: Amusingly, neither of `{hash,crypto}.Hash` is the other: `crypto.Hash` might better be called `crypto.HashAlgorithm`, but that may be 10 years too late. ;-(

**Review by itaiad200 — APPROVED** [review]

File: graveler/sstable/manager.go (hunk 1 above)
Comment (nopcoder) [reply graveler/sstable/manager.go:19]: guess they like it simple - I liked the fact that my `int` can create an hash instance

**Comment by nopcoder** [comment]
> Good catch!
> 
> A unit test cannot easily detect issues like this (loss of catalog dedupe). Wondering how to automate testing this in an ongoing manner. Best I can see is to run specific import scenarios and track disk usage.

Think I can write something that the race detector will catch.

**Pull request merged**